src/Sarif.Multitool.Library/MergeCommand.cs

```diff
@@ -117,7 +121,7 @@ public int Run(MergeOptions mergeOptions)
                     mergedLog.SchemaUri = mergedLog.Version.ConvertToSchemaUri();
 
                     FileSystem.DirectoryCreateDirectory(outputDirectory);
-                    outputFilePath = Path.Combine(outputDirectory, GetOutputFileName(_options, key));
+                    outputFilePath = Path.Combine(outputDirectory, ReplaceInvalidCharInFileName(GetOutputFileName(_options, key), "."));
```

Comment:
>ReplaceInvalidCharInFileName [](http://example.com/codeflow?start=67&length=28)

If PerRule split strategy is specified, it creates a SarifLog file per rule, the file name is {ruleId}_merged.sarif.

If the ruleId is a sub rule Id, e.g. `SEC105/001`, it contains invalid file name char '/' and will cause saving the log file failed.

This is the fix for this issue.

Reply:
Where is the unit test for this issue? Did I miss it?

Reply:
dedicated unit tests added. pls review.